lib/modules.nix

```diff
@@ -397,7 +426,7 @@ rec {
           };
         in parentFile: parentKey: initialModules: args: collectResults (imap1 (n: x:
           let
-            module = loadModule args parentFile "${parentKey}:anon-${toString n}" x;
+            module = checkModule (loadModule args parentFile "${parentKey}:anon-${toString n}" x);
```

Comment:
I think it's cleaner to use `let in` bindings in e.g. `loadModule` instead of wrapping with another function call here. This way all the processing is together and not split apart. I think functions should only be used when they have more than one use.

Reply:
> the raw module is still in scope

Good point. Maybe something like this would be nicer though:

```nix
loadModule = lib.flip lib.pipe [
  (m: ...)
  (m: ...)
]
```

Anyways, I think it's fine as-is for now